### PR TITLE
Feature/rest api get owned api list

### LINKED ISSUE #2773: [FIWARE] REST-API: User must get the list of owned APIs
Now a user can get the list of all APIs or with the specific APIs

### Goal
- Create endpoint to return all managed APIs for this user. It should use header "X-User-id"

### LINKED ISSUE #2790: REST API: GET /apis, parameter organization needs fixing
Matter sot be fixed:
### Parameter name
* in swagger document parameter name is ```organization```, however in code there are prepared for parameter name ```organizationApi```.
More descriptive name were ```organizationId```
### Functionality
Without any parameter all APIs are returned in response message.
* in case parameter organizationId is given and Organization is found, only APIs related to that organization are returned. This is correct.
* In case parameter is given, but organization is not found, all APIs are returned, which is erroneous. Expected response is an error message ```Bad query parameters```

### PR DESCRIPTION
**Pull request by matleppa**
Closes #2773 
Closes #2790 

REST API for APIs.
- new parameter for requesting APIs managed by requestor
- fixes in parameter Organization

### CODE REVIEW

File: apis/server/api.js [hunk 1]
```diff
@@ -58,13 +58,29 @@ ApiV1.addCollection(Apis, {
...
+   Parameters are optional and combinations or parameters can be used.
+
+   Example call:
+
+       GET /apis?limit=200&amp;&managedApis=true
+
+   Result: returns maximum of 200 APIs which are managed by equesting user.
+
+
+   Note! When using parameter managedAPIS, the requestor user ID is read
```
Comment (brylie) [inline apis/server/api.js]: Should be `managedAPIs`. Case sensitive.

File: apis/server/api.js [hunk 2]
```diff
@@ -92,21 +108,47 @@ ApiV1.addCollection(Apis, {
...
+          // Reaquestor User ID needed in header
+          const requestorId = this.request.headers['x-user-id'];
+
+          // Response with error in case requestorId is missing
+          if (!requestorId) {
+            return {
+              statusCode: 400,
+              body: {
+                status: 'Fail',
+                message: 'Bad query parameters. Managing User ID missing, expected in header.',
```
Comment (brylie) [inline apis/server/api.js]: Go ahead and document the header parameter (x-user-id) in the error message. Also, determine whether the parameter is case sensitive. If so, we may want to capitalize the first letter of each 'word': X-User-Id or something similar.

File: apis/server/api.js [hunk 3]
```diff
@@ -92,21 +108,47 @@ ApiV1.addCollection(Apis, {
...
         const queryParams = this.queryParams;
-
         // Only Public APIs are available for unregistered user
         const query = { isPublic: true };
         const options = {};
 
+        // Get APIs managed by user requesting operation
+        if (queryParams.managedAPIs) {
+          // Reaquestor User ID needed in header
+          const requestorId = this.request.headers['x-user-id'];
```
Comment (brylie) [inline apis/server/api.js]: Lets **keep the variable name consistent** with the parameter name. In this code, we first call something by one name ('user id') and then by another name ('requestor id') and later by a third name ('manager id'). 

Inconsistent code is hard to follow.

File: apis/server/api.js (hunk 2 above)
Comment (matleppa) [inline apis/server/api.js]: When header parameter is read in API, it can be referred only with lower case letters.
In Postman the parameter name is not case sensitive. However, in swagger login, the parameter name is capitalized.
So, I choose to write it in documentation by capitalizing the first letters of each word.

File: apis/server/api.js [hunk 4]
```diff
@@ -58,13 +58,29 @@ ApiV1.addCollection(Apis, {
...
+   from X-User-Id field in message header.
+        `,
+
         parameters: [
           ApiV1.swagger.params.optionalSearch,
           ApiV1.swagger.params.organizationApi,
           ApiV1.swagger.params.skip,
           ApiV1.swagger.params.limit,
           ApiV1.swagger.params.lifecycle,
+          ApiV1.swagger.params.managedApis,
```
Comment (brylie) [inline apis/server/api.js]: I still see a bit of inconsistency with this parameter. Choose one of the following and try to use it consistently, unless there is some specific reason not to (in which case add a comment):

- managedApis
- managedAPIs

File: apis/server/api.js [hunk 5]
```diff
@@ -92,21 +108,47 @@ ApiV1.addCollection(Apis, {
...
         const queryParams = this.queryParams;
-
         // Only Public APIs are available for unregistered user
         const query = { isPublic: true };
         const options = {};
 
+        // Get APIs managed by user requesting operation
+        if (queryParams.managedAPIs) {
+          // Get Manager ID from header
+          const managerId = this.request.headers['x-user-id'];
```
Comment (brylie) [inline apis/server/api.js]: Remember consistency. Is it `X-User-Id` or `X-User-id`?

Comment (matleppa) [inline apis/server/api.js]: In program code it is x-user-id, because JS seems to accept it only without any capitalization of letters. In documentation X-User-Id, according to way Swagger document uses the parameter name.
